pandas/tests/series/test_constructors.py

```diff
@@ -1513,6 +1514,22 @@ def test_constructor_sparse_datetime64(self, values):
         expected = Series(arr)
         tm.assert_series_equal(result, expected)
 
+    def test_constructor_datetimelike_to_sparse_object(self):
+        arr = np.arange(3, dtype="i8").view("M8[D]").astype("M8[ns]")
+        arr2 = arr.view("m8[ns]")
+
+        ser = Series(arr, dtype="Sparse[object]")
+        assert isinstance(ser[0], Timestamp)
```

Comment:
why wouldn't these be Sparse[datetime64[ns]] ? (and equiv for timedelta). do we actually have a usecase for embedding non-object dtypes here? I think this something that we should limit (as we do coercion on non-sparse things if they are datetimelike already this is a big change).

Reply:
I'm not sure i understand the question.  are you asking why L1521 isnt `ser = Series(arr, dtype="Sparse[datetime64[ns]]")`?

Looking at the non-sparse, case `Series(arr, dtype=object)[0]` returns a pydatetime object, so i guess its OK that `Sparse[object]` does the same.

Really the motivation is to make astype_nansafe with dt64 data behave the same as DatetimeArray.astype and Block._astype so we can share more.